app/jobs/application_job.rb

```diff
@@ -31,4 +31,17 @@ class ApplicationJob < ActiveJob::Base
   # retry_on ActiveRecord::Deadlocked
   # Most jobs are safe to ignore if the underlying records are no longer available
   # discard_on ActiveJob::DeserializationError
+
+  # Raises if the metadata is incomplete
+  # @param [Hash<Symbol => String>] metadata
+  # @option metadata [String] :checksum_md5
+  # @option metadata [String] :size
```

Comment:
is size here an `Integer`?

Reply:
since it's all getting passed in by the job params, i'd expect all of the params to be Strings (to be converted back to other types as needed).

Reply:
yeah in a later commit, this is changed to
`   # @option metadata [Integer] :size `